src/backends/torch/torchgraphbackend.cc

```diff
@@ -67,6 +67,47 @@ namespace dd
     finalize(dimint);
   }
 
+  bool TorchGraphBackend::extractable(std::string extract_layer)
+  {
+    for (BaseGraph::Vertex v : _sortedVars)
+      if (varname(v) == extract_layer)
+        return true;
+    return false;
+  }
+
+  std::vector<std::string> TorchGraphBackend::extractable_layers() const
+  {
+    std::vector<std::string> allvars;
+    for (BaseGraph::Vertex v : _sortedVars)
+      allvars.push_back(varname(v));
+    return allvars;
+  }
+
+  torch::Tensor TorchGraphBackend::extract(torch::Tensor inputTensor,
+                                           std::string extract_layer)
+  {
+    set_input(inputTensor);
+    bool output_computed = false;
+    for (BaseGraph::Vertex o : _sortedOps)
+      {
+        std::vector<torch::Tensor> out = forward(o);
+        std::vector<BaseGraph::Vertex> outputVars = this->outputs(o);
+        for (unsigned int i = 0; i < outputVars.size(); ++i)
+          {
+            _variables[varname(outputVars[i])] = out[i];
+            if (varname(outputVars[i]) == extract_layer)
+              return out[i];
+            if (varname(outputVars[i]) == _outputname)
+              output_computed = true;
+            _variables[varname(outputVars[i])] = out[i];
+          }
+      }
+    if (!output_computed)
+      throw TorchGraphException(
+          "did not compute output, please check NN graph");
```

Comment:
Let's have a slightly more informed message here if possible. When is this supposed to occur typically, when `extract_layer` can't be found ?

Reply:
ok done